### ISSUE TITLE
[BLUEMARLIN-30]: DIN Lookalike potential issue #53

### ISSUE THREAD

**Issue by srinaginikonduru** [Futurewei-io/blue-marlin#266]
[sanjaynirmal](https://github.com/sanjaynirmal)

Hello @jimmylao,
I was wondering that whether doing operations on negative and positive sampling would exclude some of the device IDs out from training data ?
If it does then we might get an error at the time of ctr generation for those DIDs which were not present during the training.
Wanted to know your opinion on this.

**Comment by srinaginikonduru** [comment]
[jimmylao](https://github.com/jimmylao)

@sanjaynirmal There is a placeholder in model.py for userID (or device ID) at the beginning of model.py (line 30 for placeholder, line 40 for userID embedding), however, it was not used afterwards, i.e. userID is not used in the model.
While CTR generation, the input data does need to provide DID information to follow the format of input data, however, it may not require the DID exists in training data.
In summary, the potential issue may not happen in reality.

**Comment by srinaginikonduru** [comment]
[radibnia77](https://github.com/radibnia77)  [BLUEMARLIN-30]: DIN Lookalike potential issue
